src/vs/editor/contrib/referenceSearch/browser/referenceSearch.ts

```diff
@@ -7,7 +7,7 @@
 import * as nls from 'vs/nls';
 import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
 import URI from 'vs/base/common/uri';
-import { TPromise } from 'vs/base/common/winjs.base';
+import { PPromise, TPromise } from 'vs/base/common/winjs.base';
```

Comment:
> Is Holder a pattern used by VS Code?

No, this was just the simplest way I could think of to make it work.

> Looks like a simple event emitter to me.

Event emitter seems like overkill here. We don't need to have named events, there is only one type of event.

> Don't you think it will be simpler to use the callback pattern everywhere instead of mixing with promises?

Do you think the callback pattern in my comment above is cleaner code than using PPromise here? Either way we need an object to own the progress callback (whether it is a promise or a holder).

I am inclined to leave this as is to see what the vscode people think. If they hate PPromise then I will change it. It is also possible that they would prefer using PPromise here, so I will defer to them.